pandas/util/hashing.py

```diff
@@ -1,278 +1,18 @@
-"""
-data hash pandas / numpy objects
-"""
-import itertools
+import warnings
+import sys
 
-import numpy as np
-from pandas import Series, factorize, Categorical, Index, MultiIndex
-from pandas.util import libhashing as _hash
-from pandas._libs.lib import is_bool_array
-from pandas.core.dtypes.generic import (
-    ABCIndexClass,
-    ABCSeries,
-    ABCDataFrame)
-from pandas.core.dtypes.common import (
-    is_categorical_dtype, is_numeric_dtype,
-    is_datetime64_dtype, is_timedelta64_dtype,
-    is_list_like)
+m = sys.modules['pandas.util.hashing']
+for t in ['hash_pandas_object', 'hash_array']:
 
-# 16 byte long hashing key
-_default_hash_key = '0123456789123456'
+    def outer(t=t):
 
+        def wrapper(*args, **kwargs):
+            from pandas import util
+            warnings.warn("pandas.util.hashing is deprecated and will be "
+                          "removed in a future version, import "
+                          "from pandas.util",
+                          FutureWarning, stacklevel=3)
```

Comment:
This is not needed I think, as I said above, see my recent PR here: https://github.com/pandas-dev/pandas/pull/16086/files, this was moved from pandas.tools to util.